common/src/main/java/org/apache/drill/common/expression/SchemaPath.java

```diff
@@ -220,9 +214,6 @@ public boolean contains(Object obj) {
     }
 
     SchemaPath other = (SchemaPath) obj;
-    if (rootSegment == null) {
```

Comment:
Are you sure rootSegment is not null ? If so, pls add an assert otherwise the next statement could potentially hit NPE. Also what about other.rootSegment ? Should that be checked ?